### DISCUSSION TITLE
Use a class as a nested model, still access inheriting classes fields?

### DISCUSSION THREAD

**Discussion by AdrianDeAnda** [pydantic/pydantic#9198] [Question] [answered]
Trying to do a migration from Pydantic v1 -> v2 on a large OOP codebase however we ran in to an issue where if a class has a nested model and we pass a value that is a class where the parent class is defined as a type we don't get the fields of the inheriting class when doing `model_dump` like in v1 unless we specify all the classes. Example:

Pydantic v1 behavior:
![Screenshot from 2024-04-09 20-04-32](https://github.com/pydantic/pydantic/assets/1024932/e38b6881-cc4c-42ac-aea1-7a9733473be4)

Pydantic v2 behavior, no change to types:
![Screenshot from 2024-04-09 20-06-21](https://github.com/pydantic/pydantic/assets/1024932/9587a08e-ed3e-4dee-917d-83a55fe30325)

Pydantic v2 behavior,  change to types:
![Screenshot from 2024-04-09 20-09-43](https://github.com/pydantic/pydantic/assets/1024932/b303dc64-5a94-4227-bba8-604f69101712)

Is there a configuration to keep the v1 behavior on v2 without specifying all the possible types of the classes with inheritance?

**Comment by uriyyo** [comment, marked as answer]
You can try to use `SerializeAsAny`:
```py
from pydantic import BaseModel, SerializeAsAny


class Test1(BaseModel):
    age: int


class Test2(Test1):
    name: str


class Test3(BaseModel):
    what_test: SerializeAsAny[Test1]


test_model = Test2(age=10, name="test")
model = Test3(what_test=test_model)

print(model.model_dump())
```
```py
{'what_test': {'age': 10, 'name': 'test'}}
```

**Reply by AdrianDeAnda** [reply]
Thanks, that is exactly what I was looking for.
